### PR TITLE
upgrade: deprecated if_delegate_has_method

### LINKED ISSUE #475: `if_delegate_has_method` is deprecated, use `available_if` instead
`if_delegate_has_method` is used in some places in the `aif360.sklearn` code (e.g., [reweighing.py](https://github.com/Trusted-AI/AIF360/blob/master/aif360/sklearn/preprocessing/reweighing.py)) but this is deprecated now in favor of [`available_if`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.metaestimators.available_if.html)

e.g.,
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/ensemble/_stacking.py#L660

### PR DESCRIPTION
**Pull request by AmirAflak**
Closes #475

### CODE REVIEW

File: aif360/sklearn/postprocessing/__init__.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def fit(self, X, y, sample_weight=None, **fit_params):
                                 **fit_params)
         return self
 
-    @if_delegate_has_method('postprocessor_')
+    @available_if('postprocessor_')
```
Comment (gowriaddepalli) [inline aif360/sklearn/postprocessing/__init__.py:135]: Do we need to make any changes in the unit test?

File: aif360/sklearn/postprocessing/__init__.py [hunk 2]
```diff
@@ -175,7 +175,7 @@ def predict_proba(self, X):
         y_score = pd.DataFrame(y_score, index=X.index).squeeze('columns')
         return self.postprocessor_.predict_proba(y_score)
 
-    @if_delegate_has_method('postprocessor_')
+    @available_if('postprocessor_')
```
Comment (gowriaddepalli) [inline aif360/sklearn/postprocessing/__init__.py:178]: How do we test if this is working? Have we ran the code end to end to check if this is working?

**Review by mnagired — APPROVED** [review]
looks good, seems like straightforward replacement of `available_if` for any previous `if_delegate_has_method` lines.

**Review by mnagired — CHANGES_REQUESTED** [review]
actually, looks like all of the post-processing tests are failing with the same error:

`TypeError: 'str' object is not callable`

Could you please look into this?

**Comment by hoffmansc** [comment]
closing in favor of #511

**Pull request closed without merging**